src/thunderbots/backend_output/grsim/grsim_backend.cpp

```diff
@@ -1,11 +1,54 @@
-#include "grsim_backend.h"
+#include "backend_output/grsim/grsim_backend.h"
+#include <iostream>
+#include <optional>
 #include "ai/primitive/move_prim.h"
+#include "proto/grSim_Commands.pb.h"
+#include "proto/grSim_Packet.pb.h"
 
-GrSimBackend::GrSimBackend()
+using namespace boost::asio;
+
+GrSimBackend::GrSimBackend(std::string network_address, unsigned short port)
+    : network_address(network_address), port(port), socket(io_service)
+{
+    socket.open(ip::udp::v4());
+    remote_endpoint = ip::udp::endpoint(ip::address::from_string(network_address), port);
+}
+
+GrSimBackend::~GrSimBackend()
+{
+    socket.close();
+}
+
+void GrSimBackend::setRobotVelocities(
+    unsigned int robot_id, bool team_colour_yellow, Point velocity,
+    Angle angular_velocity)
 {
```

Comment:
We should make sure we've got unit test coverage for generation of packets to send to GrSim. It's a bit tricky to test network stuff, so we could refactor this to have a 
```
std::string GrSimBackend::getRobotVelocityPacket(
    unsigned int robot_id, bool team_colour_yellow, Point velocity,  Angle angular_velocity)
```
function and test that instead?

Reply:
I'll separate this function into one that only sends a grSim packet over the network, and one that constructs the packet from parameters. Then we can test the packet construction.